src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -82,49 +82,34 @@ public void Process(Exception exception, SentryEvent sentryEvent)
 
         internal IEnumerable<SentryException> CreateSentryException(Exception exception)
         {
-            if (exception is AggregateException ae)
-            {
-                foreach (var inner in ae.InnerExceptions.SelectMany(CreateSentryException))
-                {
-                    yield return inner;
-                }
-
-                if (!_options.KeepAggregateException)
-                {
-                    yield break;
-                }
-            }
-            else if (exception.InnerException != null)
-            {
-                foreach (var inner in CreateSentryException(exception.InnerException))
-                {
-                    yield return inner;
-                }
-            }
+            return exception.EnumerateChainedExceptions(_options)
```

Comment:
I was thinking we could refactor out this class to have a base class, so that the code that follows it doesn't need to iterate on the exceptions once again:

https://github.com/getsentry/sentry-unity/pull/683/files#diff-d7c6f55c576cbd905ba5affcc1c578fd120655cdfcdc520994c837efeeebccc2R33

Then a third time to have the original + the Sentry exception side by side:

https://github.com/getsentry/sentry-unity/pull/683/files#diff-d7c6f55c576cbd905ba5affcc1c578fd120655cdfcdc520994c837efeeebccc2R47

Reply:
@bruno-garcia i dont really understand what you are proposing here. can we sync my tuesday morning?

Reply:
I thought of re-thinking these hooks but didn't get that far, https://github.com/getsentry/sentry-dotnet/pull/1734